src/Query/Builder.php

```diff
@@ -1245,12 +1288,16 @@ protected function compileWhereYear(array $where): array
      */
     protected function compileWhereTime(array $where): array
     {
-        extract($where);
-
-        $where['operator'] = $operator;
-        $where['value'] = $value;
-
-        return $this->compileWhereBasic($where);
+        return [
+            '$expr' => [
+                '$'.$where['operator'] => [
+                    [
+                        '$dateToString' => ['date' => '$'.$where['column'], 'format' => '%H:%M:%S'],
+                    ],
+                    $where['value'],
```

Comment:
We talked about validating user input. Should we enforce a string in the format `HH:MM:SS` and permit trailing components to be omitted (e.g. allow `HH:MM` or even just `HH`)?

Reply:
Fixed with support for various time formats.